attribute_based.py

```diff
@@ -0,0 +1,133 @@
+import argparse
+from datasets import *
+import pickle
+import torchvision.transforms as transforms
+from torchvision import models
+import torch.utils.data as data
+import torch.nn.functional as F
+import numpy as np
+import torch.nn as nn
+import os
+import sys
+import cv2
+import boto3
+import imageio
+import botocore
+from PIL import Image
+from cifar_models import resnet110
+from tqdm import tqdm
+
+# 0 for aws rekognition, 1 for free cv2
+FACE_DETECT = 1
+
+# call aws rekognition for face unless info already exists from being called before
+def detect_face(filepath, info, client):
+    if filepath not in info.keys():
+        with open(filepath, 'rb') as image:
+            try:
+                response = client.detect_faces(Image={'Bytes': image.read()})
+            except botocore.exceptions.ClientError as error:
+                print(error)
+                this_image = Image.open(filepath)
+                this_image.thumbnail((500, 500))
+                this_image.save('temp.jpg')
+                with open('temp.jpg', 'rb') as im:
+                    response = client.detect_faces(Image={'Bytes': im.read()})
+            info[filepath] = response
+
+    return info
+
+#Note: attribute possible values and names are passed in with args: --attribute_values '2' --attribute_names 'male female'
```

Comment:
these should not be passed in via args, these should be defined as a parameter in datasets.py. please add to the dataset template (right now they will be set to "male" and "female", so it does interrupt current gender analysis workflow)

Reply:
if this has been fixed, please remove this comment